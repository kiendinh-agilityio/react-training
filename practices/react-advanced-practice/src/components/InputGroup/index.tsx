// Import radix ui
import { Flex } from '@radix-ui/themes';

// Import components
import { Input, Text } from '@/components/common';

// Import type
import { TextSize } from '@/types';

interface InputGroupProps {
  label?: string;
  name: string;
  type?: 'search' | 'text' | 'email' | 'password' | 'date';
  value: string;
  placeholder?: string;
  errorMessage?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur: (e: React.FocusEvent<HTMLInputElement>) => void;
}

const InputGroup = ({
  label,
  name,
  type = 'text',
  value,
  placeholder = '',
  errorMessage,
  onChange,
  onBlur,
}: InputGroupProps) => (
  <Flex direction="column" className="gap-1.5">
    <Text size={TextSize.Base} as="label" className="text-dark">
      {label}
    </Text>
    <Input
      name={name}
      type={type}
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      onBlur={onBlur}
      errorMessage={errorMessage}
    />
  </Flex>
);

export default InputGroup;
