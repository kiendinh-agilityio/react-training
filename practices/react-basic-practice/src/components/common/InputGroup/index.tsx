// Import components
import { Input } from '@/components/common';

interface InputGroupProps {
  label?: string;
  name: string;
  type: string;
  value: string;
  placeholder?: string;
  errorMessage: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur: (e: React.FocusEvent<HTMLInputElement>) => void;
}

const InputGroup = ({
  label,
  name,
  type,
  value,
  placeholder,
  errorMessage,
  onChange,
  onBlur,
}: InputGroupProps) => (
  <div className="flex flex-col gap-1.5">
    <label>{label}</label>
    <Input
      name={name}
      type={type}
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      onBlur={onBlur}
      errorMessage={errorMessage}
    />
    {errorMessage && <p className="text-danger">{errorMessage}</p>}
  </div>
);

export default InputGroup;
