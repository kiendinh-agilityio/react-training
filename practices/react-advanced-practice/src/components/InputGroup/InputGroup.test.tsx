import { render } from '@testing-library/react';
import InputGroup from '.';

describe('InputGroup Component', () => {
  const mockOnChange = jest.fn();
  const mockOnBlur = jest.fn();

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <InputGroup
        label="Email Address"
        name="email"
        type="email"
        value=""
        placeholder="Enter your email"
        onChange={mockOnChange}
        onBlur={mockOnBlur}
      />,
    );

    // Match the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
