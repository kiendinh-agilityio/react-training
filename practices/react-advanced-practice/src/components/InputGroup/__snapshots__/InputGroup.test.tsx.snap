// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rt-Flex rt-r-fd-column gap-1.5"
  >
    <label
      class="rt-Text rt-r-weight-regular text-base text-dark"
    >
      Email Address
    </label>
    <div
      class="rt-Box w-full"
    >
      <div
        class="rt-Flex rt-r-ai-center rt-r-jc-center flex justify-center items-center px-5 py-[15px] border gap-3 w-full rounded-[15px] border-input dark:bg-light"
      >
        <div
          class="rt-TextFieldRoot rt-r-size-2 rt-variant-surface w-full text-regular focus:outline-none text-dark placeholder:text-base dark:bg-dark dark:bg-light"
        >
          <input
            class="rt-reset rt-TextFieldInput"
            name="email"
            placeholder="Enter your email"
            spellcheck="false"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
